src/Compilers/CSharp/Portable/Parser/SyntaxParser.cs

```diff
@@ -1054,13 +1054,20 @@ protected TNode CheckFeatureAvailability<TNode>(TNode node, MessageID feature, b
         {
             LanguageVersion availableVersion = this.Options.LanguageVersion;
 
-            if (feature == MessageID.IDS_FeatureModuleAttrLoc)
-            {
-                // There's a special error code for this feature, so handle it separately.
-                return availableVersion >= LanguageVersion.CSharp2
-                    ? node
-                    : this.AddError(node, ErrorCode.WRN_NonECMAFeature, feature.Localize());
-            }
+            // There are special error codes for some features, so handle those separately.
+            switch (feature)
+            {
+                case MessageID.IDS_FeatureModuleAttrLoc:
+                    return availableVersion >= LanguageVersion.CSharp2
+                        ? node
+                        : this.AddError(node, ErrorCode.WRN_NonECMAFeature, feature.Localize());
+
+                case MessageID.IDS_FeatureVerbatimInterpolatedStrings:
+                    return availableVersion >= LanguageVersion.CSharp8
```

Comment:
>LanguageVersion.CSharp8 [](start = 47, length = 23)

I think this should be `feature.RequiredVersion()` and again a few lines later so the mapping from feature to version is in one place only.